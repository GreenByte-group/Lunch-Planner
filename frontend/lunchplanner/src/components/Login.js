import React from "react";
import {doLogin} from "./LoginFunctions";
import {Link, Redirect} from "react-router-dom";
import Modal from 'react-modal';
import {getUser} from "./User/UserFunctions";

class Login extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            username: "",
            password: "",
            redirectToReferrer: false,
            error: "",
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);

        this.setState = {
            modalIsOpen: false
        };

        this.openModal = this.openModal.bind(this);
        this.afterOpenModal = this.afterOpenModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }



    handleInputChange(event) {
        const target = event.target;
        const name = target.name;

        this.setState({
            [name]: target.value
        });
    }

    handleSubmit(event) {
        getUser(this.state.username, (response) => {
            console.log('getUser', response);
            this.setState({
                username: response.data.userName,
            });
            if(response.status === 200){
                doLogin(response.data.userName, this.state.password, message => {

                    if (message.type === "LOGIN_EMPTY") {
                        this.setState({
                            error: message.payload.message,
                        });
                    } else if (message.type === "LOGIN_SUCCESS") {
                        this.setState({
                            error: "",
                            redirectToReferrer: true,
                        });
                    } else if (message.type === "LOGIN_FAILED") {
                        this.setState({
                            modalIsOpen: true,
                            error: "wrong username or password",
                        });
                    }
                });
            }else{
                this.setState({
                    modalIsOpen: true,
                    error: "wrong username or password",
                });
            }
        });

        event.preventDefault();
    }

    openModal() {
        this.setState({modalIsOpen: true});
    }

    afterOpenModal() {
        // references are now sync'd and can be accessed.
        this.subtitle.style.color = '#f00';
    }

    render() {
        const { from } = this.props.location.state || { from: { pathname: "/" } };
        const { redirectToReferrer } = this.state;
        const { error } = this.state;

        if (redirectToReferrer) {
            return <Redirect to={from} />;
        }
        return (
            <div className="container">
                <h3>Login</h3>
                {(error
                        ? <>{error}</>
                        : ""
                )}
                <form className="form col-md-12 center-block" onSubmit={this.handleSubmit}>
                    <div className="form-group">
                        <input className="form-control input-lg" type="text" name="username"
                               onChange={this.handleInputChange} placeholder="User Name"/>
                    </div>
                    <div className="form-group">
                        <input className="form-control input-lg" type="password" name="password"
                               onChange={this.handleInputChange} placeholder="Password"/>
                    </div>
                    <div className="form-group">
                        <input className="btn btn-primary btn-lg btn-block" type="submit" value="Login"/>
                    </div>
                </form>

                <Link to="/register" >Register</Link>
            </div>
        )
    }
}

export default Login;