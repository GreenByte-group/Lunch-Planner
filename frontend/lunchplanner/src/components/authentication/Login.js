import React from "react";
import {checkName, doLogin} from "./LoginFunctions";
import {Redirect} from "react-router-dom";
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import {Input, InputLabel, InputAdornment} from '@material-ui/core';
import { FormControl, } from '@material-ui/core';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import Button from '@material-ui/core/Button';
import LoginIcon from '@material-ui/icons/ExitToApp';
import Modal from 'react-modal';
import {getUser} from "../User/UserFunctions";


let sushi = '/sushi.jpg';
let kebab = '/kebab.jpg';

const styles = theme => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        height: '-webkit-fill-available',
        maxHeight: '100%',
        width: '-webkit-fill-available',
        maxWidth: '100%',
    },
    root2: {
        display: 'flex',
        flexDirection: 'row',
        height: '-webkit-fill-available',
        maxHeight: '100%',
        width: '-webkit-fill-available',
        maxWidth: '100%',
    },
    description: {
        fontSize: '25px',
        lineHeight: '24px',
        textAlign: 'center',
        margin: '20px',

    },
    content: {
        justifyContent: 'space-between',
        flexGrow: 1,
        padding: '0 5px',
        height: '-webkit-fill-available',
        maxHeight: '100%',
        width: '-webkit-fill-available',
        maxWidth: '100%',
    },
    button: {
        height: '-webkit-fill-available',
        maxHeight: '50px',
        width: '-webkit-fill-available',
        maxWidth: '100%',
        color: 'white',
        fontSize: '15px',
    },
    margin: {
        margin: '20px 0px',
        width: '-webkit-fill-available',
        maxWidth: '100%',
        marginLeft: '10%',
        marginRight: '10%',
    },
    withoutLabel: {
        marginTop: theme.spacing.unit * 3,
    },
    textField: {
        flexBasis: 200,
    },
    left:{
        height: '-webkit-fill-available',
        maxHeight: '100%',
        width: '-webkit-fill-available',
        maxWidth: '25%',
        float: 'left',
        overflow: 'hidden',
        backgroundImage: 'url(' + sushi + ')',
        backgroundSize: 'cover',
        backgroundPosition: 'center center',
        backgroundRepeat: 'no-repeat',

    },
    right: {
        height: '-webkit-fill-available',
        maxHeight: '100%',
        width: '-webkit-fill-available',
        maxWidth: '25%',
        float: 'right',
        overflowX: 'hidden',
        overflowY: 'hidden' ,
        backgroundImage: 'url(' + kebab + ')',
        backgroundSize: 'cover',
        backgroundPosition: 'center center',
        backgroundRepeat: 'no-repeat',
    },
    logo: {
        display: 'flex',
        position: 'relative',
        height: '-webkit-fill-available',
        maxHeight: '200px',
        width: '-webkit-fill-available',
        maxWidth: '200px',
        marginTop: '15px',
        marginLeft: 'auto',
        marginRight: 'auto',

    },

});

const customStyles = {
    content : {
        top                   : '50%',
        left                  : '50%',
        right                 : 'auto',
        bottom                : 'auto',
        marginRight           : '-50%',
        transform             : 'translate(-50%, -50%)'
    }
};

class Login extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            username: "",
            password: "",
            redirectToReferrer: false,
            error: "",
            modalIsOpen: false,

        };


        this.openModal = this.openModal.bind(this);
        this.afterOpenModal = this.afterOpenModal.bind(this);
        this.closeModal = this.closeModal.bind(this);

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }



    handleInputChange(event) {
        const target = event.target;
        const name = target.id;

        this.setState({
            [name]: target.value
        });
    }
    openModal() {
        this.setState({modalIsOpen: true});
    }

    afterOpenModal() {
        // references are now sync'd and can be accessed.
        this.subtitle.style.color = '#f00';
    }

    closeModal() {
        this.setState({modalIsOpen: false});
    }

    handleSubmit(event) {

         getUser(this.state.username, (response) => {
           console.log('getUser', response);
           this.setState({
               username: response.data.userName,
           });
           if(response.status === 200){
               console.log('res',response.data.userName);
               if(response.data.userName === undefined){
                   this.setState({
                       modalIsOpen: true,
                       error: "wrong username or password",
                       username: "",
                       password: "",
                   });
               }else{
                   doLogin(response.data.userName, this.state.password, message => {

                       if (message.type === "LOGIN_EMPTY") {
                           this.setState({
                               error: message.payload.message,
                               username: "",
                               password: "",
                           });
                       } else if (message.type === "LOGIN_SUCCESS") {
                           this.setState({
                               error: "",
                               redirectToReferrer: true,
                           });
                       } else if (message.type === "LOGIN_FAILED") {
                           this.setState({
                               modalIsOpen: true,
                               error: "wrong username or password",
                               username: "",
                               password: "",
                           });
                       }
                   })
               }
              ;
           }else{
               this.setState({
                   modalIsOpen: true,
                   error: "wrong username or password",
               });
           }
        });

        if(event)
            event.preventDefault();
    }

    handleChange = prop => event => {
        this.setState({ [prop]: event.target.value });
    };

    handleMouseDownPassword = event => {
        event.preventDefault();
    };

    handleClickShowPassword = () => {
        this.setState({ showPassword: !this.state.showPassword });
    };

    keyPress = (e) => {
        if(e.keyCode === 13){
            this.handleSubmit();
        }
    };

    render() {
        const { classes } = this.props;
        const { from } = /*this.props.location.state || */{ from: { pathname: "/" } };
        const { redirectToReferrer } = this.state;
        const { error } = this.state;

        if (redirectToReferrer) {
            return <Redirect to={from} />;
        }

        let disabled = true;
        if(this.state.username && this.state.password)
            disabled = false;

        return (
            <div className={classes.root2}>
                <div className={classes.left}/>
                <div className={classes.root}>
                    <div className={classes.content}>
                        {(error
                                ? <Modal
                                    isOpen={this.state.modalIsOpen}
                                // onAfterOpen={this.afterOpenModal}
                                onRequestClose={this.closeModal}
                                style={customStyles}
                                contentLabel="Example Modal"
                            >{error}</Modal>
                            : ""
                    )}
                    <p className={classes.description}>
                        Welcome back. <br/>
                    </p>
                    <img label="logoLunchplanner" src="/foodastic.png" className={classes.logo}/>
                    <FormControl
                        fullWidth
                        className={classes.margin}
                        aria-describedby="weight-helper-text"
                    >
                        <InputLabel>{<p style={{fontSize: '18px'}}>Username</p>}</InputLabel>
                        <Input
                            style={{fontSize: '20px'}}
                            id="username"
                            value={this.state.username}
                            onChange={this.handleInputChange}
                            inputProps={{
                                'aria-label': 'Username',
                            }}

                        />
                    </FormControl>
                    <FormControl
                        fullWidth
                        className={classes.margin}
                    >
                        <InputLabel htmlFor="adornment-password">{<p style={{fontSize: '18px'}}>Password</p>}</InputLabel>
                        <Input
                            id="password"
                            style={{fontSize: '20px'}}
                            type={this.state.showPassword ? 'text' : 'password'}
                            value={this.state.password}
                            onChange={this.handleInputChange}
                            onKeyDown={this.keyPress}
                            endAdornment={
                                <InputAdornment position="end">
                                    <IconButton
                                        aria-label="Toggle password visibility"
                                        onClick={this.handleClickShowPassword}
                                        onMouseDown={this.handleMouseDownPassword}
                                    >
                                        {this.state.showPassword ? <VisibilityOff /> : <Visibility />}
                                    </IconButton>
                                </InputAdornment>
                            }
                        />
                    </FormControl>
                </div>
                <Button disabled={disabled} fullWidth variant="raised" color="secondary" className={classes.button} onClick={this.handleSubmit}>
                    <LoginIcon color={"#FFFFF"}/>LOGIN
                </Button>
            </div>
                <div className={classes.right}/>
            </div>
        );
    }
}

Login.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(Login);
