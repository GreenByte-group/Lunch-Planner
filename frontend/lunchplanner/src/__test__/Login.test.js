import React from 'react';
import Login from '../components/Login';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
   const tree = renderer.create(
       <MemoryRouter>
           <Login location={{state: "/"}} />
       </MemoryRouter>
   ).toJSON();

   expect(tree).toMatchSnapshot();
});