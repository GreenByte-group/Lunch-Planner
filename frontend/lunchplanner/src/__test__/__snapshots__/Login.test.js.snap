// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h3>
    Login
  </h3>
  
  <form
    className="form col-md-12 center-block"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="username"
        onChange={[Function]}
        placeholder="User Name"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control input-lg"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn btn-primary btn-lg btn-block"
        type="submit"
        value="Login"
      />
    </div>
  </form>
  <a
    href="/register"
    onClick={[Function]}
  >
    Register
  </a>
</div>
`;
